src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -323,6 +323,12 @@ function VisEditor($rootScope, $scope, $route, timefilter, AppState, $window, kb
       if (id) {
         notify.info('Saved Visualization "' + savedVis.title + '"');
         if ($scope.isAddToDashMode()) {
+          // We need to manually set the last url for this app so navigating back to visualize will open the saved
+          // visualization, not the unsaved one.
+          const savedVisualizationUrl =
+            kbnUrl.eval(`${kbnBaseUrl}#${VisualizeConstants.EDIT_PATH}/{{id}}`, { id: savedVis.id });
+          chrome.trackSubUrlForApp('kibana:visualize', savedVisualizationUrl);
```

Comment:
Ah, good point.  So, maybe there is something obvious I am missing, but if I just push a new url on the browser history it won't trigger the new sub url tracking for the link.  I played around with $location.path(newurl).replace()  but that didn't seem to do anything.  Also, I'm not sure I want to replace the current url, just add a new one, since the way it would normally work is that two back button presses will get you back to the unsaved url.

I'm guessing the chrome suburl tracking needs the page to reload before storing the last sub url?